modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js

```diff
@@ -111,6 +111,71 @@ export function handleAction(
 
 		window.open(url);
 	}
+	else if (target === 'download') {
+		event.preventDefault();
+
+		setLoading(true);
+
+		let filename = '';
+
+		executeAsyncAction(url, method)
+			.then((response) => {
+				const disposition = response.headers.get('Content-Disposition');
+
+				if (disposition && disposition.indexOf('attachment') !== -1) {
+
+					// Captures the filename from the Content-Disposition header.
+					// For example, "attachment; filename="Filename.json" will capture
+					// "Filename.json".
+
+					const filenameRegex = /filename[^;=\n]*=((['"]).*?\2|[^;\n]*)/;
+
+					const matchesRegex = filenameRegex.exec(disposition);
+
+					if (matchesRegex !== null && matchesRegex[1]) {
+						filename = matchesRegex[1].replace(/['"]/g, '');
+					}
+				}
+
+				return response.blob();
+			})
+			.then((responseBlob) => {
+				const downloadUrl = URL.createObjectURL(responseBlob);
+				const downloadElement = document.createElement('a');
+
+				if (
+					filename &&
+					typeof downloadElement.download !== 'undefined'
+				) {
+					downloadElement.download = filename;
+					downloadElement.href = downloadUrl;
+
+					document.body.appendChild(downloadElement);
+
+					downloadElement.click();
+				}
+				else {
+					window.location.href = downloadUrl;
+				}
```

Comment:
Is there a difference in these download methods and `window.open`, when it comes to setting authentication headers? I don't think either of them is setting it. We're setting [auth header](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/fetch.es.js#L29) in `fetch` util, used in the first request, but that doesn't have anything to do with the download itself. 

If this was a normal DXP resource URL, I would expect this header would be set by using `<liferay-portlet:resourceURL>`, `liferayPortletResponse.createResourceURL`, or a similar util. I don't know how this is set with REST API, but I would expect it to be set in backend, before passed to a frontend component.

An an unrelated note, `typeof downloadElement.download !== 'undefined'` seems to be a backwards compatibility check for IE. We don't support IE, so we don't need this check.

Reply:
For better context, our SXP Element export URL is defined here: https://github.com/liferay/liferay-portal/blob/a3a615adb4df920316561cec06558676c7e94377/modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/SXPElementResourceImpl.java#L86.

Using `executeAsyncAction` sets the auth header with `fetch`, right? I'm unsure what you mean? 😅  

We can take out the IE check, sounds good.

Reply:
> Using `executeAsyncAction` sets the auth header with `fetch`, right? I'm unsure what you mean?

We are not downloading with `executeAsyncAction` / `fetch`.

In this PR, we are making two requests:
1. The ajax request that returns download resource URL and has file name in response header. This is executed with `fetch`, has auth header set, all good. Except for the fact that this logic is app-specific and should not be in a general-purpose component.
2. The download request. This is executed with `window.open()`, with `click()` on a link with a `download` attribute, or with `window.location.href`. I don't see why any of these methods would add auth header to the request, unless there is some weird interaction I am missing (like code being executed within `then` block of a `fetch`).

Reply:
I see, performing the ajax request and then the download request is too specific to what search experiences wants to do. It was just the solution that worked for the export API URL without issues of authentication and content type.

Would it be fine to request this option instead, then we'd be able to place this logic in our own module?
"If not, we in FI can make it a high priority to add support for propsTransformer, so you can add any behavior you wish by overriding onClick on the action item."

Reply:
OK. I'll try to make in the next two days, then I'm on 2 weeks PTO :slightly_smiling_face:  At the very least, I'll analyze deep enough to see if there are any pitfalls, and pass the task to somebody else. It's a feature we need to have in any case, it's just a matter of priorities when to do it. /cc @dsanz

Reply:
Thank you, Marko, that's so greatly appreciated!!